src/include/txn.i

```diff
@@ -1251,10 +1251,10 @@ __wt_txn_update_check(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDATE
     if (!rollback && upd == NULL && cbt != NULL && CUR2BT(cbt)->type != BTREE_COL_FIX &&
       cbt->ins == NULL) {
         __wt_read_cell_time_window(cbt, cbt->ref, &tw);
-        if (tw.stop_txn != WT_TXN_MAX && tw.stop_ts != WT_TS_MAX)
-            rollback = !__wt_txn_visible(session, tw.stop_txn, tw.stop_ts);
+        if (WT_TIME_WINDOW_HAS_STOP(&tw))
+            rollback = !__wt_txn_tw_stop_visible(session, &tw);
```

Comment:
__wt_txn_tw_stop_visible already checks if WT_TIME_WINDOW_HAS_STOP. Additionally this is a minor logic change as the `&&` becomes an `||`.

Reply:
The previous code
`if (tw.stop_txn != WT_TXN_MAX && tw.stop_ts != WT_TS_MAX)`
is not correct. Luckily we don't have the case that stop_txn is max. 
Though `__wt_txn_tw_stop_visible ` already checks that, we still need the check otherwise rollback will be set to true for the time window without a stop time point. The compiler will remove the redundant check and I keep it this way to make the code clean.

Reply:
Alright makes sense